dask/bag/core.py

```diff
@@ -1134,7 +1134,7 @@ def partition(grouper, sequence, npartitions, p, nelements=2**20):
         d2 = defaultdict(list)
         for k, v in d.items():
             d2[abs(hash(k)) % npartitions].extend(v)
-        p.append(d2)
+        p.append(d2, fsync=True)
```

Comment:
I don't. I was able to reproduce every time when using the distributed scheduler on a single machine, but that seems a bit overkill for a test here. I can also reproduce using `multiprocessing.set_start_method('spawn')`, but that can't be set after startup, so it wouldn't work for a test. Perhaps the issue has to do with sharing file handles between different processes, which may(?) behave differently if they're sibling processes instead of separate.